### DISCUSSION TITLE
Issue when adding noise?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#74] [Q&A]
Sorry I am not sure if this should go here or in issues. Feel free to move it where it fits better.

I wanted to study the noise. So I synthesized a spectrum following the example:

spe1 = rc2.spectrum.from_delta_lines(deltas={500:1000, 700:1500}, xcal=lambda x: x*3 -1000, nbins=1000)
spe2 = spe1.convolve(lineshape='gaussian', sigma=2)
spe3 = spe2.add_baseline(n_freq=50, amplitude=10, pedestal=10)
spe4 = spe3.add_poisson_noise(scale=10) # scale multiplies the squared root of the maximum of the spectrum 
(spe4-spe3).plot()

Now, If I do spe4-spe3, this should give me the value of the noise in every point of the spectrum, right? The thing is that, when I look into it, I find that the noise at the position of the peaks is clearly higher than at other positions. Am I doing something wrong? is there a bug somewhere?

Thank you for your time in advance,
Nico

**Comment by georgievgeorgi** [comment]
This is an expected behavior since `add_poisson_noise` adds noise proportional to the amplitude of the initial spectrum scaled with `scale` parameter.

```math
\tilde{Y_i} = \hat{Y_i} + \underbrace{RandomNumber_i}_{\in N\left(\sigma=\sqrt{\hat{Y_i}*scale}\right)},
```
where:
- $\hat{Y_i}$ is the $i$-th value of the initial spectrum
- $\tilde{Y_i}$ is the $i$-th value of the noisyfied spectrum
- $N$ Gaussian distribution around zero with a given $\sigma$
- `scale` -- the provided parameter to `add_poisson_noise`

The motivation is the counting statistics -- $\Delta N \propto \sqrt{N}$

This approximation to the noise might be not really applicable in the context of CCD measurements because there are different sources of noise -- Johnson noise for example

**Reply by nicocopez** [reply]
Thank you for the answer Georgi. Honestly, I myself am not sure wether can be applicable here or not or to what extend. You probably know better. Let me study and think about it and I will come back to you with more questions. I have in mind a couple of references that I would like to check.  
We stay in touch,
nico

**Comment by georgievgeorgi** [comment]
We have an option to add gaussian noise. Here is a little documentation about that method:
https://h2020charisma.github.io/ramanchada2/ramanchada2/spectrum.html#Spectrum.add_gaussian_noise
